packages/node/src/integrations/http.ts

```diff
@@ -192,13 +192,28 @@ function _createWrappedRequestMethodFactory(
 
       const scope = getCurrentHub().getScope();
 
+      const spanData: Record<string, string> = {
+        url: requestUrl,
+        method: requestOptions.method || 'GET',
+      };
+      if (requestOptions.hash) {
+        // strip leading "#"
+        spanData['http.fragment'] = requestOptions.hash.substring(1);
+      }
+      if (requestOptions.search) {
+        // strip leading "?"
+        spanData['http.query'] = requestOptions.search.substring(1);
+      }
+
+      // TODO potential breaking change... shouldCreateSpanForRequest no longer has access to query + fragment, is that a problem?
```

Comment:
Calling this out for code reviewer. There are no tests for `shouldCreateSpan` that rely on query or fragment, but an end user could be relying on it today. Options:
- continue as is here. maybe add the breaking change to release notes
- refactor this PR so that we have a `rawRequestUrl` for `shouldCreateSpan ` + `shouldAttachTraceData`, and only use the sanitized request URL for spans + breadcrumbs

Reply:
👍 on it